### PR TITLE
Improved `loadRewards` to be able to specify a specific bucket to load.

### PR DESCRIPTION
**Pull request by renesansz**
Pull Request changes:
* `loadRewards` will now accept a string parameter to specify a bucket
* Updated README

### CODE REVIEW

File: src/android/io/branch/BranchSDK.java [hunk 1]
```diff
@@ -802,19 +803,29 @@ public void onRegisterViewFinished(boolean registered, BranchError error) {
...
         // Listener that implements BranchReferralStateChangedListener for loadRewards
         @Override
         public void onStateChanged(boolean changed, BranchError error) {
             if (error == null) {
 
-                int credits = instance.getCredits();
-
+                int credits = 0;
+                
+                if (this._bucket.length()) {
```
Comment (aaustin) [inline src/android/io/branch/BranchSDK.java]: if bucket is null, this will crash.

File: src/android/io/branch/BranchSDK.java [hunk 2]
```diff
@@ -802,19 +803,29 @@ public void onRegisterViewFinished(boolean registered, BranchError error) {
     protected class LoadRewardsListener implements Branch.BranchReferralStateChangedListener
     {
         private CallbackContext _callbackContext;
+        private Branch _instance;
+        private String _bucket;
 
         // Constructor that takes in a required callbackContext object
-        public LoadRewardsListener(CallbackContext callbackContext) {
+        public LoadRewardsListener(CallbackContext callbackContext, Branch instance, String bucket) {
```
Comment (renesansz) [inline src/android/io/branch/BranchSDK.java]: hmm, but I made sure that it won't have a null value.

File: src/android/io/branch/BranchSDK.java [hunk 3]
```diff
@@ -1127,7 +1138,9 @@ public void run() {
                     } else if (this.action.equals("logout")) {
                         logout(this.callbackContext);
                     } else if (this.action.equals("loadRewards")) {
-                        loadRewards(this.callbackContext);
+                        if (this.args.length() == 1) {
+                            loadRewards(this.callbackContext, this.args.getString(0));
```
Comment (aaustin) [inline src/android/io/branch/BranchSDK.java]: If there are existing implementations calling without 'bucket', it'll break and never return. Please make this argument optional.

Comment (renesansz) [inline src/android/io/branch/BranchSDK.java]: @aaustin ,  added overload method for `loadRewards` optional parameter bucket string.

**Comment by aaustin** [comment]
looks great! 👍

**Pull request merged**